src/SFA.DAS.EmployerIncentives.Functions.PaymentsProcess/Activities/CollectionPeriodInProgress.cs

```diff
@@ -0,0 +1,30 @@
+﻿using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.DurableTask;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.EmployerIncentives.Abstractions.Queries;
+using System.Threading.Tasks;
+using SFA.DAS.EmployerIncentives.Queries.CollectionCalendar.CollectionPeriodInProgress;
+
+namespace SFA.DAS.EmployerIncentives.Functions.PaymentsProcess
+{
+    public class CollectionPeriodInProgress
+    {
+        private readonly IQueryDispatcher _queryDispatcher;
+        private readonly ILogger<CollectionPeriodInProgress> _logger;
+
+        public CollectionPeriodInProgress(IQueryDispatcher queryDispatcher, ILogger<CollectionPeriodInProgress> logger)
+        {
+            _queryDispatcher = queryDispatcher;
+            _logger = logger;
+        }
+
+        [FunctionName(nameof(CollectionPeriodInProgress))]
+        public async Task<bool> Get([ActivityTrigger] object input)
+        {
+            _logger.LogInformation("Getting Collection Period In Progress");
+            var isInProgress = (await _queryDispatcher.Send<CollectionPeriodInProgressRequest, CollectionPeriodInProgressResponse>(new CollectionPeriodInProgressRequest())).IsInProgress;
```

Comment:
Why put the business functionality in the request?  Would it not be better to just use the exiting GetActiveCollectionPeriodQueryHandler (via GetActiveCollectionPeriod ativity) and add the InProgress property in the response?  Then this very specific query is not required.